include/dbstructure.php

```diff
@@ -242,7 +242,7 @@ function update_structure($verbose, $action, $tables=null, $definition=null) {
 				if (!isset($database[$name]["fields"][$fieldname])) {
 					$sql2=db_add_table_field($fieldname, $parameters);
 					if ($sql3 == "") {
-						$sql3 = "ALTER TABLE `".$temp_name."` ".$sql2;
+						$sql3 = "ALTER" . $ignore . " TABLE `".$temp_name."` ".$sql2;
```

Comment:
Maybe move the "version detection code block from around line 180 to somewhere generic and encapsulate it into a function/method, then rewrite this (and many parts in `update.php`) to call it?

If I search for only **ALTER TABLE**, I can find 177 occurrances.